### PR TITLE
Permission system

### PR DESCRIPTION
**Pull request by sgratzl**
start with client side permission system for datasets and unified the user session management

### CODE REVIEW

**Comment by mstreit** [comment]
Is this ready to be merged? If yes, can @thinkh go ahead and merge it?

File: src/security.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+ */
+export function isLoggedIn() {
+  return retrieve('logged_in') === true;
+}
+
+/**
+ * stores the given user information
+ * @param user
+ */
+export function login(user: {name: string}) {
```
Comment (thinkh) [inline src/security.ts]: Stupid question: Am I logged in, if I import/call this function with any user name? Or even easier, if I add these variables to the store I have access to the full functionality? How/When is the permission checked?

(I just want to make sure that there is no obvious security hole, when we implementing a sophisticated permission system.)

**Review by thinkh — COMMENTED** [review]
@sgratzl Could you please write in short how to use the permission system or point to an example code.

File: src/security.ts (hunk 1 above)
Comment (sgratzl) [reply src/security.ts]: what means ` I have access to the full functionality`?

**Comment by thinkh** [comment]
> @sgratzl Could you please write in short how to use the permission system or point to an example code.

Could you please provide some example code how to use this implementation?

**Pull request merged**